nixos/ombi: add baseUrl option

Elfahor opened the pull request:
The base URL GUI option of Ombi is known to be broken, which is very inconvenient for reverse proxying with subfolders. The suggested method (in Ombi's documentation) is to pass the `--baseurl` command-line parameter.

## Description of changes

This commit thus adds an option to set this parameter to the NixOS Ombi module.

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Elfahor committed:
nixos/ombi: add baseUrl option

The base URL GUI option of Ombi is known to be broken, which is very inconvenient
for reverse proxying with subfolders. The suggested method is to pass the
--baseurl command-line parameter.
This commit thus adds an option to set this parameter to the NixOS Ombi module.

nu-nu-ko commented on nixos/modules/services/misc/ombi.nix:
```diff
@@ -31,6 +31,12 @@ in {
+      baseUrl = mkOption {
+        type = types.str;
+        default = "/";
+        description = lib.mdDoc "The base URL for the Ombi web interface.";
```
`lib.mdDoc` is now just an alias and can be safely removed everywhere.
see https://github.com/NixOS/nixpkgs/commit/d36f950d405369abca5144b14a201bc25c4a080a and https://github.com/NixOS/nixpkgs/pull/237557

Elfahor replied:
Do you want me to remove it from all the options in the module, or only for the one I added?

Elfahor replied:
I removed everywhere in [106710b](https://github.com/NixOS/nixpkgs/pull/284311/commits/106710b5e1d01d871a527ef17bdd17cdc485df96)

nu-nu-ko commented on nixos/modules/services/misc/ombi.nix:
```diff
@@ -59,7 +65,7 @@ in {
         User = cfg.user;
         Group = cfg.group;
-        ExecStart = "${pkgs.ombi}/bin/Ombi --storage '${cfg.dataDir}' --host 'http://*:${toString cfg.port}'";
+        ExecStart = "${pkgs.ombi}/bin/Ombi --storage '${cfg.dataDir}' --host 'http://*:${toString cfg.port}' --baseurl ${cfg.baseUrl}";
```
`${lib.getExe pkgs.ombi}` would be nice.

Elfahor replied:
Done in [883d86f](https://github.com/NixOS/nixpkgs/pull/284311/commits/883d86fefaa8833e220bb0a9ebc4a41e05c79c26)

Elfahor committed:
Merge branch 'NixOS:master' into ombi-baseurl

Elfahor committed:
nixos/ombi: remove lib.mdDoc

Elfahor committed:
nixos/ombi: use lib.getExe

Elfahor commented:
I have successfully tested these changes on a server.

fsnkty reviewed (approved):
changes seem fine, id be more descriptive in the enable option about what the service is though..

also seen in the module unrelated to these changes, would be nice to see `systemd.tmpfiles.settings` used insteead.

Elfahor commented:
Should I make these changes in this PR or open an other one?

On February 18, 2024 1:23:16 AM GMT+01:00, nuko ***@***.***> wrote:
>@nu-nu-ko approved this pull request.
>
>changes seem fine, id be more descriptive in the enable option about what the service is though..
>
>also seen in the module unrelated to these changes, would be nice to see `systemd.tmpfiles.settings` used insteead.
>
>
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/NixOS/nixpkgs/pull/284311#pullrequestreview-1887050994
>You are receiving this because you authored the thread.
>
>Message ID: ***@***.***>

nu-nu-ko commented:
> Should I make these changes in this PR or open an other one?

Id assume same PR different commit but am unsure, either way neither are obligations of yours, just random things id like to see personally.

I'd wait for someone more familiar with contributing to get back to you on that one if you wish to make the changes.

Elfahor commented:
I'm open to making them. I just wish that what I already added be merged, since I added the new `baseUrl` option because I needed it in the first place.

On February 20, 2024 5:42:04 AM GMT+01:00, nuko ***@***.***> wrote:
>> Should I make these changes in this PR or open an other one?
>
>Id assume same PR different commit but am unsure, either way neither are obligations of yours, just random things id like to see personally.
>
>I'd wait for someone more familiar with contributing to get back to you on that one if you wish to make the changes.
>
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/NixOS/nixpkgs/pull/284311#issuecomment-1953478937
>You are receiving this because you authored the thread.
>
>Message ID: ***@***.***>

eclairevoyant reviewed (changes requested):
Hi, thanks for the contribution! Please rebase to drop the merge commit, as well as to drop the mdDoc removal commit (this was handled treewide). (Of course, the new option shouldn't use lib.mdDoc either, as it was removed from nixpkgs.)

Otherwise the changes look good and I'll be happy to merge afterwards.

Elfahor commented:
Unfortunately, I don't think this is going to work. Closing.
